src/esp32/entry.rs

```diff
@@ -1,77 +1,71 @@
-    let ret = unsafe {
-        xTaskCreatePinnedToCore(
-            Some(client_entry),                // C ABI compatible entry function
-            CLIENT_TASK.as_ptr() as *const i8, // task name
-            8192 * 3,                          // stack size
-            ip as *mut c_void,                 // pass ip as argument
-            20,                                // priority (low)
-            &mut hnd,                          // we don't store the handle
-            0,                                 // run it on core 0
-        )
+pub fn serve_web(
+    app_config: AppClientConfig,
+    tls_server_config: Esp32TlsServerConfig,
+    robot: Option<LocalRobot>,
+    _ip: Ipv4Addr,
+    webrtc_certificate: WebRtcCertificate,
+) {
+    let robot = Arc::new(Mutex::new(robot.unwrap()));
+
+    let mut client_connector = Esp32Tls::new_client();
+    let exec = Esp32Executor::new();
+    let mdns = Esp32Mdns::new("".to_string()).unwrap();
+
+    let robot_cfg = {
+        let cloned_exec = exec.clone();
+        let conn = client_connector.open_ssl_context(None).unwrap();
+        let conn = Esp32Stream::TLSStream(Box::new(conn));
+        let grpc_client =
+            Box::new(GrpcClient::new(conn, cloned_exec, "https://app.viam.com:443").unwrap());
+
+        let builder = AppClientBuilder::new(grpc_client, app_config.clone());
+
+        let mut client = builder.build().unwrap();
+
+        client.get_config().unwrap()
     };
-    if ret != 1 {
-        return Err(anyhow::anyhow!("wasn't able to create the client task"));
-    }
-    log::error!("got handle {:?}", hnd);
-    Ok(hnd)
-}
 
-/// client main loop
-fn clientloop(config: AppClientConfig) -> Result<()> {
-    let mut tls = Box::new(Esp32Tls::new_client());
-    let conn = tls.open_ssl_context(None)?;
-    let conn = Esp32Stream::TLSStream(Box::new(conn));
-    let executor = Esp32Executor::new();
+    let address: SocketAddr = "0.0.0.0:12346".parse().unwrap();
+    let tls = Box::new(Esp32Tls::new_server(&tls_server_config));
+    let tls_listener = Esp32Listener::new(address.into(), Some(tls)).unwrap();
 
-    let grpc_client = GrpcClient::new(conn, executor, "https://app.viam.com:443")?;
+    let webrtc_certificate = Rc::new(webrtc_certificate);
+    let dtls = Esp32DtlsBuilder::new(webrtc_certificate.clone());
 
-    let mut _app_client = AppClientBuilder::new(grpc_client, config).build();
+    let cloned_exec = exec.clone();
 
-    log::error!("current task handle {:?}", unsafe {
-        xTaskGetCurrentTaskHandle()
-    });
-    Ok(())
-}
+    let webrtc = Box::new(WebRtcConfiguration::new(
+        webrtc_certificate,
+        dtls,
+        client_connector,
+        exec.clone(),
+        app_config,
+    ));
+
+    let mut srv = Box::new(
+        ViamServerBuilder::new(mdns, tls_listener, webrtc, cloned_exec, 12346)
+            .build(&robot_cfg)
+            .unwrap(),
+    );
 
-/// C compatible entry function
```

Comment:
could you explain a bit why we needed these extern functions and things from `esp_idf_sys` and why we don't need them now? I still need to read up on FreeRTOS